src/fides/api/util/tcf/ac_string.py

```diff
@@ -0,0 +1,193 @@
+        + ".".join([str(vendor_id) for vendor_id in sorted(ac_vendor_consents)])
+    )
+
+
+def build_fides_string(tc_str: Optional[str], ac_str: Optional[str]) -> str:
+    """Concatenate a TC string and an AC string into a 'fides string' representation, to represent
+    both in a single string"""
+    if tc_str is None:
+        tc_str = ""
+
+    if not ac_str:
+        return tc_str
+
+    return f"{tc_str}{FIDES_SEPARATOR}{ac_str}"
+
+
+def split_fides_string(fides_str: Optional[str]) -> Tuple[Optional[str], Optional[str]]:
+    """Split a Fides String into separate TC Strings and AC Strings
+
+    Run some upfront validation here on the Fides String format as a whole, and the AC string section.
+    TC string section validation is more complex and happens elsewhere.
+    """
+    if not fides_str:
+        return None, None
+
+    split_str = fides_str.split(FIDES_SEPARATOR)
+    tc_str = split_str[0] or None
+
+    ac_str = None
+
+    if len(split_str) > 2:
+        raise DecodeFidesStringError("Unexpected Fides String format")
+
+    if len(split_str) == 2:
+        ac_str = split_str[1]
+
+    validate_ac_string_format(ac_str)
+    return tc_str, ac_str
+
+
+def validate_ac_string_format(ac_str: Optional[str]) -> None:
+    """Run some preliminary validation checks on the AC str"""
+    if not ac_str:
+        return
+
+    if not ac_str.startswith(SPECIFICATION_VERSION_NUMBER + SEPARATOR_SYMBOL):
+        raise DecodeFidesStringError("Unexpected AC String format")
+
+    ac_str = ac_str.lstrip(SPECIFICATION_VERSION_NUMBER).lstrip(SEPARATOR_SYMBOL)
+
+    try:
+        [int(vendor_id) for vendor_id in ac_str.split(".")]
+    except ValueError:
+        raise DecodeFidesStringError("Unexpected AC String format")
+
+
+def _ac_str_to_universal_vendor_id_list(ac_str: Optional[str]) -> List[str]:
+    """Helper to convert an AC string into a list of universal ac vendor ids
+
+    Used when saving preferences from an AC string
+    """
+    if not ac_str:
+        return []
+
+    validate_ac_string_format(ac_str)
+
+    ac_str = ac_str.lstrip(SPECIFICATION_VERSION_NUMBER).lstrip(SEPARATOR_SYMBOL)
+    vendor_ids: List[str] = ac_str.split(".")
+
+    universal_ac_vendor_ids: List[str] = [
+        AC_PREFIX + vendor_id for vendor_id in vendor_ids
+    ]
+
+    return universal_ac_vendor_ids
+
+
+def _convert_ac_strings_to_fides_preferences(
+    current_ac_str: Optional[str],
+    accept_all_ac_str: Optional[str],
+) -> List[TCFVendorSave]:
+    """Helper to convert an AC string into a list of corresponding vendor preferences, for saving preferences
+    from an AC string.
+
+    AC strings only encode opt-ins, so by itself, this doesn't give us explicit opt-outs. We compare the passed-in AC
+    String with the accept-all AC string built at runtime against our current datamap.
+
+    Vendors in both strings are saved as an opt-in.  AC vendors in the datamap, but not in the supplied AC string,
+    get persisted as an explicit opt-out.
```

Comment:
Note I use the same strategy here when translating an AC string to fides preferences, as I did for TC strings.  AC strings only encode opt in, but I compare to the datamap directly to also be able to persist explicit opt-outs.